util/crc32c.cc

```diff
@@ -375,10 +378,7 @@ static bool isSSE42() {
 #ifndef HAVE_SSE42
   return false;
 #elif defined(__GNUC__) && defined(__x86_64__) && !defined(IOS_CROSS_COMPILE)
-  uint32_t c_;
-  uint32_t d_;
-  __asm__("cpuid" : "=c"(c_), "=d"(d_) : "a"(1) : "ebx");
-  return c_ & (1U << 20);  // copied from CpuId.h in Folly.
+  return __builtin_cpu_supports("sse4.2");
```

Comment:
thanks for your review. no, i cannot be sure about this. =/

clang supports this feature, see https://reviews.llvm.org/rL240994. but it was 2015. while clang has been supported C++11 since 3.3 back to 2013.

GCC 4.8+ supports this builtin function. hence GCC on mingw/cygwin does.   rocksdb uses C++11. and C++11 is well supported by GCC since 4.8. also it's a simple change only relies on GCC. so it's safe on GCC side.

in short, i will drop this change as it might break the SSE42 optimization on older clang versions.